File: SoalB/PHP/Hardware.php
<?php

class Hardware extends Product
{
    /* class hardware adalah child dari class product */

    /* atribut privat */
    private $brand,
        $model;

    /* membuat constructor */
    public function __construct()
    {
        $this->brand = "";
        $this->model = "";
    }

    /* Get-Set Methods brand */
    public function setBrand($brand)
    {
        $this->brand = $brand;
    }

    public function getBrand()
    {
        return $this->brand;
    }

    /* Get-Set Methods model */
    public function setModel($model)
    {
        $this->model = $model;
    }

    public function getModel()
    {
        return $this->model;
    }

    /* membuat desturctor */
    public function __destruct()
    {
    }
}

File: SoalB/C++/hardware.cpp
#include "product.cpp"

class hardware : public product
{
    /* class hardware adalah child dari class product */

    /* atribut privat */
private:
    string model;
    string brand;

public:
    /* membuat constructor */
    hardware()
    {
    }

    /* Get-Set Methods model */
    void setModel(string model)
    {
        this->model = model;
    }
    string getModel()
    {
        return this->model;
    }

    /* Get-Set Methods brand */
    void setBrand(string brand)
    {
        this->brand = brand;
    }
    string getBrand()
    {
        return this->brand;
    }

    /* membuat destructor */
    ~hardware()
    {
    }
};